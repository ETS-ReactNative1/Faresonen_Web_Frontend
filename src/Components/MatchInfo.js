import React, { Component } from 'react';
import { Item } from 'semantic-ui-react';
import moment from 'moment-with-locales-es6';
import axios from 'axios';
moment.locale('no');

class MatchInfo extends Component {

	constructor(props) {
		super(props);
		this.state = {
			matchName: props.match.name,
			channel: '',
			stadium: '',
			startDate: new Date(props.match.starttime).toLocaleDateString(),
			startTime: new Date(props.match.starttime).toLocaleTimeString()
		}
		
		axios.all([this.getChannelAndStadium(props.match)])
			.then(axios.spread((data) => {
			if(data.channel) {
				data.channel.then(channel => this.setState({ channel: channel.data.name }));	
			} else {
				this.setState({
					channel: 'Kanal ikke klar'
				})
			}

			if(data.stadium) {
				data.stadium.then(stadium => this.setState({ stadium: stadium.data.name }));	
			} else {
				this.setState({
					stadium: 'Stadion klar'
				})
			}
			
			
		}));
	}

	getChannelAndStadium(match) {
		return { 
			channel: this.getChannel(match),
			stadium: this.getStadium(match) 
		}
	}

	getChannel(match) {
		if(match.channel) {
			return axios.get(match.channel['@uri']);	
		} else {
			return null;
		}
		
	}

	getStadium(match) {
		if(match.stadium) {
			return axios.get(match.stadium['@uri'])	
		} else {
			return null;
		}
		
	}



	render() {
		return (
				<Item>
					<Item.Content>
						<Item.Header>{this.state.matchName}, {this.state.stadium}</Item.Header>
						<Item.Meta>{this.state.startDate} - Avspark kl. {this.state.startTime} p√• {this.state.channel}</Item.Meta>
					</Item.Content>
				</Item>
			)
	}
}

export default MatchInfo;