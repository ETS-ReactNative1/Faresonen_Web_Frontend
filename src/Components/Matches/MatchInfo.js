import React, { Component } from "react";
import PropTypes from "prop-types";
import { Item } from "semantic-ui-react";
import axios from "axios";
import moment from "moment";
import "moment/locale/nb";
import tools from "../../Tools/tools";
import events from "../../Tools/events";

class MatchInfo extends Component {
  static formatRefereeName(ref) {
    if (ref.firstname && ref.lastname) {
      return `${ref.firstname} ${ref.lastname}`.trim();
    } else if (ref.firstname) {
      return `${ref.firstname}`.trim();
    } else if (ref.lastname) {
      return `${ref.lastname}`.trim();
    }
    throw new Error("Referee must have either firstname or lastname property");
  }

  constructor(props) {
    super(props);
    this.state = {
      matchName: props.match.name,
      channel: "",
      stadium: "",
      startDate: moment(props.match.starttime).format("DD/MM YYYY"),
      startTime: tools.getTime(props.match.starttime),
      referee: ""
    };
  }

  componentDidMount() {
    if (this.props.match.referee) {
      this.getReferee(this.props.match.referee["@uri"]);
    }
    axios.all([tools.getChannelAndStadium(this.props.match)]).then(
      axios.spread(data => {
        if (data.channel) {
          data.channel.then(channel =>
            this.setState({ channel: channel.data.name })
          );
        } else {
          this.setState({
            channel: "Kanal ikke klar"
          });
        }

        if (data.stadium) {
          data.stadium.then(stadium =>
            this.setState({ stadium: stadium.data.name })
          );
        } else {
          this.setState({
            stadium: "Stadion klar"
          });
        }
      })
    );
  }

  getReferee(refUri) {
    axios.get(refUri).then(data => {
      this.setState({
        referee: MatchInfo.formatRefereeName(data.data)
      });
    });
  }

  render() {
    let header = "";
    const { status } = this.props.match;
    if (status && events.postponed.includes(status["@uri"])) {
      header = (
        <Item.Header>
          UTSATT: {this.state.matchName}, {this.state.stadium}
        </Item.Header>
      );
    } else {
      header = (
        <Item.Header>
          {this.state.matchName}, {this.state.stadium}
        </Item.Header>
      );
    }
    return (
      <Item>
        <Item.Content>
          {header}
          <Item.Meta>
            {this.state.startDate} - Avspark kl. {this.state.startTime} p√•{" "}
            {this.state.channel}
          </Item.Meta>
          {this.state.referee && (
            <Item.Meta className="float-right">
              Dommer: {this.state.referee}
            </Item.Meta>
          )}
        </Item.Content>
      </Item>
    );
  }
}

MatchInfo.propTypes = {
  match: PropTypes.shape({
    name: PropTypes.string.isRequired,
    starttime: PropTypes.string.isRequired,
    referee: PropTypes.shape({ "@uri": PropTypes.string.isRequired }),
    status: PropTypes.shape({ "@uri": PropTypes.string.isRequired })
  }).isRequired
};

export default MatchInfo;
