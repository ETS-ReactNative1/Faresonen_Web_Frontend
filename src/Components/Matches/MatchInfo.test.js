import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { Item } from "semantic-ui-react";
import MatchInfo from "./MatchInfo";

describe("<MatchInfo />", () => {
  it("Should render correctly", () => {
    const tree = renderer.create(
      <MatchInfo
        matchName="Man. United - Liverpool"
        stadium="Old Trafford"
        startDate="2018-02-05T13:45:00+02.00"
        startTime="13.45"
        channel="TV 2 Sport Premium"
        referee="Martin Atkinson"
      />
    );
    expect(tree).toMatchSnapshot();
  });

  it("Should render header as Utsatt (delayed) if the status is delayed", () => {
    const elem = shallow(
      <MatchInfo
        matchName="Man. United - Liverpool"
        stadium="Old Trafford"
        startDate="2018-02-05T13:45:00+02.00"
        startTime="13.45"
        channel="TV 2 Sport Premium"
        referee="Martin Atkinson"
        status={{
          "@uri": "http://api.tv2.no/sport/resources/statuses/2/"
        }}
      />
    );
    const header = elem.find(Item.Header);
    expect(header.html()).toEqual(
      '<div class="header">UTSATT: Man. United - Liverpool, Old Trafford</div>'
    );
  });

  it("Should render a normal header if no status is provided", () => {
    const elem = shallow(
      <MatchInfo
        matchName="Man. United - Liverpool"
        stadium="Old Trafford"
        startDate="2018-02-05T13:45:00+02.00"
        startTime="13.45"
        channel="TV 2 Sport Premium"
        referee="Martin Atkinson"
      />
    );
    const header = elem.find(Item.Header);
    expect(header.html()).toEqual(
      '<div class="header">Man. United - Liverpool, Old Trafford</div>'
    );
  });
});
