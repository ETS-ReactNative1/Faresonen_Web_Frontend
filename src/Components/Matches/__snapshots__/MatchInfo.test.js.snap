// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MatchInfo /> Should render correctly 1`] = `
<div
  className="item"
>
  <div
    className="content"
  >
    <div
      className="header"
    >
      Man. United - Liverpool
      , 
      Old Trafford
    </div>
    <div
      className="meta"
    >
      2018-02-05T13:45:00+02.00
       - Avspark kl.
      13.45
       p√•
      TV 2 Sport Premium
    </div>
    <div
      className="meta float-right"
    >
      Dommer:
      Martin Atkinson
    </div>
  </div>
</div>
`;
